client/src/unifyfs-sysio.c

```diff
@@ -1440,6 +1787,65 @@ ssize_t UNIFYFS_WRAP(pwrite64)(int fd, const void* buf, size_t count,
     }
 }
 
+int UNIFYFS_WRAP(fchdir)(int fd)
+{
+    /* determine whether we should intercept this path */
+    if (unifyfs_intercept_fd(&fd)) {
+        /* lookup file id for file descriptor */
+        int fid = unifyfs_get_fid_from_fd(fd);
+        if (fid < 0) {
+            errno = EBADF;
+            return -1;
+        }
+
+        /* lookup path for fd */
+        const char* path = unifyfs_path_from_fid(fid);
+
+        /* TODO: test that path is not a file? */
+
+        /* we're happy to change into any directory in unifyfs
+         * should we check that we don't change into a file at least? */
+        if (unifyfs_cwd != NULL) {
+            free(unifyfs_cwd);
+        }
+        unifyfs_cwd = strdup(path);
+        return 0;
+    } else {
+        MAP_OR_FAIL(fchdir);
+        int ret = UNIFYFS_REAL(fchdir)(fd);
+
+        /* if the change dir was successful,
+         * update our current working direcotry */
```

Comment:
Oops, same typo here also:
```suggestion
         * update our current working directory */
```

Reply:
Thanks.